src/Native/System.Native/pal_process.cpp

```diff
@@ -17,6 +17,12 @@
 #include <syslog.h>
 #include <unistd.h>
 
+#if HAVE_LIBPROC
+#include <libproc.h>
+#include <sys/proc_info.h>
+#include <sys/resource.h>
+#endif
```

Comment:
As noted by @saper in a related thread (#2070); at some point, would it make sense to use [`sysctl`](https://en.wikipedia.org/wiki/Sysctl) interface explicitly for `__APPLE__` and `BSD`? (note that `BSD` macro is defined in `<sys/param.h>` which is available in all Unices but this macro is defined only in BSD OS's - [source](http://nadeausoftware.com/articles/2012/01/c_c_tip_how_use_compiler_predefined_macros_detect_operating_system))